tests/functional/test_install_reqs.py

```diff
@@ -10,6 +10,12 @@
 from tests.lib.local_repos import local_checkout
 
 
+if os.name == 'posix':
```

Comment:
this is a toxic destructive test - if ever any user with a preconfigured configfile runs it, they will face a lost configuration, and potentially hours or days of figuring why something they knew worked is suddenly broken

Reply:
The code in current `master` does the same.

https://github.com/pypa/pip/blob/404838abcca467648180b358598c597b74d568c9/tests/functional/test_install_reqs.py#L232-L236

So yes, it points to user HOME, but I thought that test framework should mock that, no?

Reply:
@RonnyPfannschmidt so any ideas how to change that?

Reply:
This is fine since we have an `isolate` fixture to isolate the tests from the actual system, which is used when `virtualenv` are used.

Reply:
AFAIK, by the time we isolate, it's too late, since this is done during collection.